pkg/ring/lifecycler.go

```diff
@@ -456,6 +477,18 @@ func (i *Lifecycler) initRing(ctx context.Context) error {
 
 		ingesterDesc, ok := ringDesc.Ingesters[i.ID]
 		if !ok {
+			// We use the tokens from the file only if it does not exist in the ring yet.
+			if len(tokensFromFile) > 0 {
+				level.Info(util.Logger).Log("msg", "adding tokens from file", "num_tokens", len(tokensFromFile))
+				if len(tokensFromFile) >= i.cfg.NumTokens {
+					tokensFromFile = tokensFromFile[:i.cfg.NumTokens]
```

Comment:
Interesting, we should do this if we find tokens in the ring too no? Why are we only doing this for `tokensFromFile`? Also, while decreasing the tokens operation is super rare, if we do this, as the tokens are sorted, I can imagine how we will be ignoring everything at the end and cause huge imbalances if I am not wrong.

Reply:
Can you not handle this here and open an issue for it rather, as not handling is the what we do currently if I am not wrong?

Reply:
Sounds fine, will do that